src/vs/workbench/contrib/chat/browser/chatDragAndDrop.ts

```diff
@@ -489,16 +496,25 @@ function symbolId(resource: URI, range?: IRange): string {
 	return resource.fsPath + rangePart;
 }
 
-// function extractImageAttributes(html: string): { src: string; alt?: string } {
-// 	const imgTagRegex = /<img[^>]+src=["']([^"']+)["'][^>]*>/;
-// 	const altRegex = /alt=["']([^"']+)["']/;
+function extractImageAttributes(e: DragEvent): { src: string; alt?: string } {
+	const html = e.dataTransfer?.getData('text/html');
```

Comment:
Can you remind me which browser was putting the image only in `text/html`? Maybe there's another way to get the data in this case?

Reply:
everything except safari puts stuff into `text/html` nicely. in safari, the only drag events are `text/uri-list` and `text/plain` which made it tricky

Reply:
For drag and drop, there's also the  `files` object on the `DropEvent`:

```ts
dropEvent.dataTransfer.files[0].bytes()
```

I'd check that first. It won't work for every browser / use case, so then you can fallback to `text/uri-list` and finding uris with supported extensions 

Hopefully you can avoid using `text/html`

Reply:
iirc i tried that, but `files` was always empty in every `DragEvent` :(

edge:
![Screenshot 2025-03-20 at 4 55 31 PM](https://github.com/user-attachments/assets/07e63e9e-4d73-4e4d-8792-814e1cd9b3a7)

chrome case url:
![Screenshot 2025-03-20 at 4 58 57 PM](https://github.com/user-attachments/assets/52f18aad-76b8-4ceb-b96d-7408470a07df)

chrome case where we are dragging base64:
![Screenshot 2025-03-20 at 4 57 46 PM](https://github.com/user-attachments/assets/d193cb0a-2042-4d25-b394-4679ec85eda0)

safari:
![Screenshot 2025-03-20 at 4 56 39 PM](https://github.com/user-attachments/assets/8eb10949-f94b-4bd5-a01a-b0923ff93503)

Reply:
I think this is actually a MacOS + chromium bug (see discussion in https://issues.chromium.org/issues/40805071). Seems to work for me on Windows when I tested (it also works if dropping into Safari)

Worth using the `files` object if it's there since it really simplifies things, while also keeping the fallback. I'd just use the `text/uri-list` though instead of trying to parse the html